### ISSUE TITLE
Some games run without trainer

### ISSUE THREAD

**Issue by stimulusprime** [SheffeyG/CheatDeck#19]
Hey! Maybe this question was already solved, though I can’t find a solution. It seems like some games do not allow to run something parallel. Like Elden Ring or Hollow Knight. They just run without trainers. I’ve tried everything, can’t get, what’s the problem.

**Comment by Mte90** [comment]
I have similar issue with Ghostrunner.
Maybe adding a notification when the trainer is executed so the user knows when is started can be something.

**Comment by Mte90** [comment]
Reading the plugin code just set some variable inside steam doesn't execute the trainer itself like Steam Tinker Launch.
The variables `PROTON_REMOTE_DEBUG_CMD` and `PRESSURE_VESSEL_FILESYSTEMS_RW` so it isn't possible that this plugin shows an alert.

Reading on https://github.com/SheffeyG/CheatDeck/blob/a5804031a543dcf851e2a426a4fa19295da4fc89/src/views/Normal.tsx#L51 those parameters should be configured inside steam.

Trying to understanding how works I see that a similar plugin use the same method but check if the parameters are really saved https://github.com/Tormak9970/bash-shortcuts/blob/cf7ee7831bbd076272ec593703c34ca19a961ddb/src/lib/controllers/SteamController.ts#L235
Basically wait a bit and does a check https://github.com/Tormak9970/bash-shortcuts/blob/cf7ee7831bbd076272ec593703c34ca19a961ddb/src/lib/controllers/SteamController.ts#L91

So the only thing is to check if steam saves those parameters, I have to see in the game setting next time on my steam deck.

**Comment by SheffeyG** [comment]
Please make sure the the developer mode and cef debugger is enabled, and there is always some compatibility issues for some games，but in my case，the hollow knight trainer v2021.06.13 works fine

**Comment by Mte90** [comment]
I will do that check but there isn't a way to do an alert about those if they are enabled?

**Comment by SheffeyG** [comment]
@Mte90 Yes, we can do the check when we click save buttom, but I don't think the reason trainer does not run with the game is the options are not saved.

**Comment by Mte90** [comment]
> @Mte90 Yes, we can do the check when we click save buttom, but I don't think the reason trainer does not run with the game is the options are not saved.

So the only way is to check the proton log for that game and see what says, there is a variable for that `PROTON_LOG`?

I don't know where to find a window executables that run everywhere and just open a messagebox that says "Yes I am working".

**Comment by Mte90** [comment]
I tried and CEF is enabled on system and on Decky and dev options are enabled in decky and steam deck too.

**Comment by Adravil** [comment]
Bumping up the topic, because facing the same problem.
* Tested the latest available Soulstone Survivors Fling trainer -- trainer doesn't launch.
* The same trainer with Hollow Knight launches without problems.
* Older versions of the trainer for the Soulstone Survivors don't launch with Soulstone Survivors too.
* Trainer for other game I've recently used (Hot Wheels) doesn't launch with Soulstone Survivors too.

So the problem seem to be game-related 🤔

**Comment by Richm246** [comment]
I am trying it with Retro City Rampage DX and it won't launch Aurora, but Sleeping Dogs: Definite Edition is fine

**Comment by GrizzlyFoxCat** [comment]
> Please make sure the the developer mode and cef debugger is enabled, and there is always some compatibility issues for some games，but in my case，the hollow knight trainer v2021.06.13 works fine

Where would I download this trainer? I downloaded the fling [Hollow.Knight.v1.5.68-v1.5.80.Plus.11.Trainer-FLiNG](https://flingtrainer.com/downloads/SKOTSgoKVZ_vR4n1MWfY_Q,,) and it doesn't load.

**Comment by gauravcanon** [comment]
Hey @Mte90, a bit late to the party but I had the same issue with Ghostrunner and finally got it working. Here’s what fixed it for me:

1. Grabbed the **Fling Trainer** file.
2. Renamed it to something short (just makes life easier).
3. The latest build (`Ghostrunner.v1.0-v20220628`) didn’t work for me, so I rolled back to an older one (`Ghostrunner.v1.0-v20211207`).
4. In the game’s properties (⚙️), forced it to run with **DirectX 12**.
5. Rebooted.

After that, it ran fine. I was seeing exactly the same problem you described before.
